setup.py

```diff
@@ -24,13 +24,47 @@
 import subprocess
 import textwrap
 import warnings
+import versioneer
+import builtins
 
+# This is a bit hackish: we are setting a global variable so that the main
+# numpy __init__ can detect if it is being loaded by the setup routine, to
+# avoid attempting to load components that aren't built yet.  While ugly, it's
+# a lot more robust than what was previously being used.
+builtins.__NUMPY_SETUP__ = True
+
+# Needed for backwards code compatibility below and in some CI scripts.
+FULLVERSION = versioneer.get_version()
+ISRELEASED = 'dev' not in FULLVERSION
+MAJOR, MINOR, MICRO = FULLVERSION.split('.')[:3]
+VERSION = f"{MAJOR}.{MINOR}.{MICRO}"
```

Comment:
We can't use `f-strings` in setup.py, it makes users of Python 3.5 unable to use `pip` to install old numpy versions.

Reply:
Fixed.